src/openrct2-win/openrct2-win.cpp

```diff
@@ -20,37 +20,86 @@
 #include <iterator>
 #include <openrct2-ui/Ui.h>
 #include <openrct2/core/String.hpp>
+#include <shellapi.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string>
 #include <vector>
 
-static std::vector<std::string> GetCommandLineArgs(int argc, wchar_t** argvW);
+static std::vector<std::string> GetCommandLineArgs(const wchar_t* cmdLine);
+static bool CreateConsole();
 
 /**
- * Windows entry point to OpenRCT2 with a console window using a traditional C main function.
+ * Windows entry point to OpenRCT2 for a GUI application, calling to the traditional C main function.
  */
-int wmain(int argc, wchar_t** argvW, [[maybe_unused]] wchar_t* envp)
+int WINAPI WinMain(
+    [[maybe_unused]] HINSTANCE hInstance, [[maybe_unused]] HINSTANCE hPrevInstance, [[maybe_unused]] LPSTR lpCmdLine,
+    [[maybe_unused]] int nShowCmd)
```

Comment:
```suggestion
int WINAPI WinMain(HINSTANCE, HINSTANCE, LPSTR, int)
```
They are unconditionally unused so just removing the variable makes more sense.

Reply:
I removed `[[maybe_unused]]` and kept the parameter names, but commented out. I feel this is the most readable, warning-free way to handle it.

Reply:
Fair enough